### PR TITLE
Add "refused to come back" appointment cancel reason

### PR DESCRIPTION
**Pull request by siddh1004**
https://app.shortcut.com/simpledotorg/story/10033/include-new-result-of-call-option-refused-to-come-back-app-side

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/removeoverdueappointment/RemoveOverdueEffectHandler.kt [hunk 1]
```diff
@@ -54,6 +55,16 @@ class RemoveOverdueEffectHandler @AssistedInject constructor(
     }
   }
 
+  private fun markPatientAsRefusedToComeBack(): ObservableTransformer<MarkPatientAsRefusedToComeBack, RemoveOverdueEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .observeOn(schedulersProvider.io())
+          .doOnNext { patientRepository.updatePatientStatusToMigrated(it.patientId) }
```
Comment (msasikanth) [inline app/src/main/java/org/simple/clinic/removeoverdueappointment/RemoveOverdueEffectHandler.kt:62]: Did we confirm it with server/product folks that patients who are refusing to comeback are marked as migrated?

**Review by msasikanth — CHANGES_REQUESTED** [review]
Left couple of comments, PTAL

**Comment by msasikanth** [comment]
Title suggestion

```suggestion
Add "refused to come back" appointment cancel reason
```

File: app/src/main/java/org/simple/clinic/removeoverdueappointment/RemoveOverdueEffectHandler.kt (hunk 1 above)
Comment (siddh1004) [reply app/src/main/java/org/simple/clinic/removeoverdueappointment/RemoveOverdueEffectHandler.kt:62]: Yes. Please see the conversation over [here](https://simpledotorg.slack.com/archives/C013HS4N9U3/p1677745160716859?thread_ts=1674733894.784699&cid=C013HS4N9U3)

Comment (siddh1004) [reply app/src/main/java/org/simple/clinic/removeoverdueappointment/RemoveOverdueEffectHandler.kt:62]: Yes. Please see the conversation over [here](https://simpledotorg.slack.com/archives/C013HS4N9U3/p1677745160716859?thread_ts=1674733894.784699&cid=C013HS4N9U3)

**Review by msasikanth — DISMISSED** [review]

**Review by msasikanth — APPROVED** [review]

**Pull request merged**